python/oneseismic/scan/scanners.py

```diff
@@ -206,3 +206,133 @@ def report(self):
             'sample-value-max' : self.maxsample
         }
 
+
+def is_closer(target, point, other):
+    def manhattan(x, y): return sum(map(lambda i, j: abs(i - j), x, y))
+
+    return manhattan(other, target) > manhattan(point, target)
+
+
+class GeoScanner(Scanner):
+    """ Find the mapping parameters from UTM coordinates to line numbers
+
+    Finds the UTM coordinates and line numbers of "trace0", the trace in the
+    (il_min, xl_min) corner of the cube, and the (x, y) steps in inline and
+    crossline direction.
+
+    Note that this assumes uniform inline and crossline spacing.
+
+    """
+    def __init__(self, il_word, xl_word, endian):
+        super().__init__(endian)
+        self.xl_word = xl_word
+        self.il_word = il_word
+
+        self.iline_min = None
+        self.iline_max = None
+        self.xline_min = None
+        self.xline_max = None
+
+        self.p0_linenos = None
+        self.p1_linenos = None
+        self.p2_linenos = None
+
+        self.p0 = None
+        self.p1 = None
+        self.p2 = None
+
+    def get_cdp(self, header, word):
+        scale = header[segyio.su.scalco]
+        # SEG-Y specifies that a scaling of 0 should be interpreted as identity
+        if scale == 0:
+            scale = 1
+
+        cdp = header[word]
+        if scale > 0:
+            return cdp * scale
+        else:
+            return cdp / -scale
+
+    def scan_trace_header(self, header):
+        iline = self.intp.parse(header[self.il_word])
+        xline = self.intp.parse(header[self.xl_word])
+        cdpx = self.get_cdp(header, segyio.su.cdpx)
+        cdpy = self.get_cdp(header, segyio.su.cdpy)
```

Comment:
Not 100% sure about that to be quite honest... The header will always be there containing something, but it might not be sensible data, so I'm not quite sure how we would check for this (Perhaps it is set to 0 if not recorded?). I suggest we assume that it is for now and look into it later.